src/transformState.js

```diff
@@ -4,8 +4,33 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case `removeProperties`:
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        // Do i need blank line between cycle and break?
+        break;
+
+      case `clear`:
+        for (const key in state) {
+          delete state[key];
+        }
+
+        break;
+
+      default:
+    }
+  }
```

Comment:
add to `default` some action. Like `return 'error'`, or `throw new Error('actions is not correct');`